core-rust/core-api-server/src/core_api/handlers/lts/state_account_resource_balance.rs

```diff
@@ -0,0 +1,84 @@
+use crate::core_api::*;
+use radix_engine::types::{Decimal, ResourceAddress};
+use state_manager::{
+    jni::state_manager::ActualStateManager,
+    query::{dump_component_state, VaultData},
+    store::traits::QueryableProofStore,
+};
+
+#[tracing::instrument(skip(state), err(Debug))]
+pub(crate) async fn handle_rc_state_account_fungible_resource_balance(
+    state: State<CoreApiState>,
+    request: Json<models::LtsStateAccountFungibleResourceBalanceRequest>,
+) -> Result<Json<models::LtsStateAccountFungibleResourceBalanceResponse>, ResponseError<()>> {
+    core_api_read_handler(
+        state,
+        request,
+        handle_rc_state_account_fungible_resource_balance_internal,
+    )
+}
+
+fn handle_rc_state_account_fungible_resource_balance_internal(
+    state_manager: &ActualStateManager,
+    request: models::LtsStateAccountFungibleResourceBalanceRequest,
+) -> Result<models::LtsStateAccountFungibleResourceBalanceResponse, ResponseError<()>> {
+    assert_matching_network(&request.network, &state_manager.network)?;
+
+    if !request.account_address.starts_with("account_") {
+        return Err(client_error(
+            "Only component addresses starting with account_ currently work with this endpoint.",
+        ));
+    }
+
+    let mapping_context = MappingContext::new(&state_manager.network);
+    let extraction_context = ExtractionContext::new(&state_manager.network);
+
+    let fungible_resource_address =
+        extract_resource_address(&extraction_context, &request.fungible_resource_address)
+            .map_err(|err| err.into_response_error("fungible_resource_address"))?;
+
+    if let ResourceAddress::NonFungible(_) = fungible_resource_address {
+        return Err(client_error(
+            "The provided resource address is not a fungible resource.",
+        ));
+    }
+
+    let component_address =
+        extract_component_address(&extraction_context, &request.account_address)
+            .map_err(|err| err.into_response_error("account_address"))?;
+
+    let component_dump = dump_component_state(state_manager.store(), component_address)
```

Comment:
Not super important, but before launch we should definitely make this more efficient - by just looking up the substate path we need, direct to the single vault for this resource in the account.

Could we add a TODO here?

Reply:
If the component doesn't exist, what error do we get here?

We should ensure that:
* If the account address is a normal account address, then we return a 404
* If the account address is a virtual account address, then we return a response with 0 balance.

Note: We can tell if the account address is virtual or not by matching on the component address